src/executor/executor.cc

```diff
@@ -0,0 +1,71 @@
+#include "executor/executor.hh"
+
+namespace sim {
+
+static auto applyOffset(RegVal val, RegVal imm) {
+  int64_t res = static_cast<int64_t>(val) + signCast(imm);
+  return static_cast<RegVal>(res);
+}
```

Comment:
- Could you please provide some error checking here (asserts at least)
- I think it's better to name `std::int64_t` like `SDWord` for example, where `S` for signed and `D` for double